configure.ac

```diff
@@ -372,6 +372,7 @@ AC_CHECK_FUNCS([ \
     sched_getscheduler \
     sched_setscheduler \
     strchrnul \
+    strnlen \
```

Comment:
I'm not sure I like this, when we already have plenty of code, that requires this function without prior check …

Reply:
A quick `grep -r strnlen` shows no code in htop has been using `strnlen` before this PR.

Reply:
Update. `strnlen` was recently added in #1707.